main_withdc.py

```diff
@@ -0,0 +1,104 @@
+"""This runs main and outputs it to discord"""
+import os
+import sys
+import pickle
+import time
+import datetime
+import cv2
+import face_recognition
+import discord
+import functions
+
+CAMINDEX = 0
+SAVE_FILE = "encodings.pickle"
+FONT = cv2.FONT_HERSHEY_SIMPLEX
+FONTSCALE = 1
+TEXTCOLOUR = (255, 0, 0)
+TOLERANCE = 0.4
+if not os.path.exists(SAVE_FILE):
+    data = {"encodings":[],"names":[]}
+    print("No encoding file exists running face detection only")
+else:
+    with open(SAVE_FILE, "rb") as f:
+        data = pickle.load(f)
+    Knownencodings = data["encodings"]
+    names = data["names"]
+
+
+intents = discord.Intents.default()
+intents.message_content = True
+client = discord.Client(intents=intents)
+
+@client.event
+async def on_ready():# pylint: disable=too-many-locals,too-many-branches
+    ''' runs the face recognition here'''
+    prevmessage = ""
+    print(f"{client.user} logged in")
+
+    cap = functions.camera_connection(CAMINDEX)
+
+    while True:
+        ret , frame = cap.read()
+        if not ret:
+            print("couldnt get picture")
+            sys.exit(1)
+
+        boxes = face_recognition.face_locations(frame)
+        encodings = face_recognition.face_encodings(frame, boxes)
+        matches = []
+        foundnames = []
+        for facenum,encoding in enumerate(encodings):
+            matches = face_recognition.compare_faces(Knownencodings,encoding, tolerance = TOLERANCE)
+            #print(matches)
+            if not matches:
+                for i in boxes:
+                    matches.append(False)
+            found = "Unknown"
+            for matchnum,i in enumerate(matches):
+                top,right,bottom,left = boxes[facenum]
+                cv2.rectangle(frame, (left,top), (right,bottom) , (0,255,0), 4)
+                
+                if i:
```

Comment:
Since the output of this program is a discord message and someone isn't necessarily going to be looking at the window any more (when using this script). You could add a `DRAW_BOXES` boolean at the top of the file to optionally draw these.

If `DRAW_BOXES` is true, then do this.
IF `DRAW_BOXES` is false, then skip this step.